src/components/VideoStrip.jsx

```diff
@@ -0,0 +1,219 @@
+    }
+
+    if (prevProps.totalLength !== totalLength) {
+      this.updateCommentLists();
+    }
+  }
+
+  updateActiveComments() {
+    const { progress } = this.props;
+    const { activeComment, activeComments } = this.state;
+    let condition;
+
+    if (!activeComment) {
+      condition = true;
+    } else {
+      condition = Math.abs(progress - activeComment.timestamp) >= 2;
+    }
+
+    if (condition) {
+      for (let i = 0; i < activeComments.length; i++) {
+        const comment = activeComments[i];
+
+        if (Math.abs(comment.timestamp - progress) <= 0.1) {
+          this.setState({ activeComment: comment, showActiveComment: true });
+          break;
+        }
+
+        if (Math.abs(comment.timestamp - progress) <= 0.5) {
+          this.setState({ showActiveComment: false });
+          break;
+        }
+      }
+    }
+  }
+
+  updateCommentLists() {
+    const { comments, totalLength } = this.props;
+    const activeCommentMap = {};
+    const listedCommentMap = {};
+
+    comments.sort((c1, c2) => new Date(c1.createdAt) > new Date(c2.createdAt));
+
+    comments.forEach((comment) => {
+      if (!overlapsActiveComment(comment, activeCommentMap)) {
+        activeCommentMap[Math.round(comment.timestamp)] = comment;
+        listedCommentMap[Math.round(comment.timestamp)] = comment;
+      } else if (!overlapsPlacedIcon(comment, totalLength, listedCommentMap)) {
+        listedCommentMap[Math.round(comment.timestamp)] = comment;
+      }
+    });
+
+    const activeComments = [];
+    _.forEach(activeCommentMap, (comment) => {
+      activeComments.push(comment);
+    });
+
+    const listedComments = [];
+    _.forEach(listedCommentMap, (comment) => {
+      listedComments.push(comment);
+    });
+
+    this.setState({ activeComments, listedComments });
+  }
+
+  handleHandleDrag(event) {
+    event.preventDefault();
+
+    const { moving } = this.state;
+
+    if (!this.handle || !moving) {
+      return;
+    }
+
+    const { handleSeek, totalLength } = this.props;
+    const originalX = this.handle.getBoundingClientRect().left;
+    const drag = event.pageX - originalX;
+    const deltaSeconds = drag * totalLength / 1024;
+
+    handleSeek(deltaSeconds);
+  }
+
+  renderComment() {
+    const { activeComment, showActiveComment } = this.state;
+
+    if (!activeComment) {
+      return null;
+    }
+
+    const { username, content } = activeComment;
+
+    return (
+      <Card className={`activeComment ${showActiveComment ? 'show' : ''}`}>
+        <Card.Body>
+          <Card.Title>{`${username} said:`}</Card.Title>
+          <Card.Text>{content}</Card.Text>
+        </Card.Body>
+      </Card>
+    );
+  }
+
+  renderCommentIcons() {
+    const { totalLength } = this.props;
+    const { listedComments } = this.state;
+
+    return (
+      <div className="commentIcon">
+        {
+          listedComments.map(comment => (
+            <StripComment comment={comment} totalLength={totalLength} />
```

Comment:
Maybe not too important, but a warning for the mapping:
Each child in a list should have a unique "key" prop.